lib/chromic_pdf/template.ex

```diff
@@ -171,7 +176,7 @@ defmodule ChromicPDF.Template do
     content = Keyword.get(opts, :content, @default_content)
     header = Keyword.get(opts, :header, "")
     footer = Keyword.get(opts, :footer, "")
-    styles = do_styles(opts)
+    styles = if(opts[:style] == :custom, do: "", else: do_styles(opts))
```

Comment:
No, sorry, this is not what we want, IMO. The whole idea of `Template` is to abstract the complexity of dealing with page dimensions in CSS and in the `printToPDF` options, so that as a user I can simply say "I want A4 paper and have a header of 4cm". This is solved by prepending this custom stylesheet to the input, so if you remove the stylesheet, using `Template` at all is kind of pointless (except maybe the `:a4` -> inch conversion, but maybe we can expose that as a function instead, if its useful by itself).

If a user does not want to use this behaviour, they're free to set their own `printToPDF` options and not use `Template`. They can also construct `source_and_options` structs if they now want to set `printToPDF` options for each of their multiple inputs. We can explain this in the docs when we describe multiple inputs handling.

Last but not least, if in your specific use-case you need to set additional page margins (especially left/right), you can easily add them in your own stylesheet. That's at least how we internally use `ChromicPDF.Template`: It should set up the overall page size (ISO A4) and tell Chrome how big the header is, but otherwise don't apply any margins, as we set them in the stylesheets rendered with our templates later.

Reply:
No problem at all! 👍 
It was only a workaround to show you the status of it (I also didn't like it much)

In my case I was more worried of Top/Bottom margins as I have to play with them for different headers/footers...
but I will set individual `header_height` & `footer_height` for all the templates 👌

Reply:
Exactly :+1: Idea of `Template` is to abstract the `printToPDF` opts, as Chrome's behaviour (in particular in combination with `@page` stylesheets) is somewhat peculiar.  That's also why I named these options `header_height` and `footer_height`, to separate them from the underlying API.